docs/components/plugins/APISection.tsx

```diff
@@ -0,0 +1,194 @@
+        {parameters ? <H4>Arguments</H4> : null}
+        {parameters ? (
+          <UL>
+            {parameters?.map((p: any) => (
+              <LI key={`param-${p.name}`}>
+                <InlineCode>
+                  {p.name}: {resolveTypeName(p.type)}
+                </InlineCode>
+              </LI>
+            ))}
+          </UL>
+        ) : null}
+        {comment?.shortText ? (
+          <ReactMarkdown renderers={renderers}>{comment.shortText}</ReactMarkdown>
+        ) : null}
+        <br />
+        {comment?.returns ? <H4>Returns</H4> : null}
+        {comment?.returns ? (
+          <UL>
+            <LI returnType>
+              <InlineCode>{resolveTypeName(type)}</InlineCode>
+            </LI>
+          </UL>
+        ) : null}
+        {comment?.returns ? (
+          <ReactMarkdown renderers={renderers}>{comment.returns}</ReactMarkdown>
+        ) : null}
+        <hr />
+      </>
+    );
+  });
+
+const renderEnum = ({ name, children, comment }: any): JSX.Element => {
+  return (
+    <>
+      <H3>
+        <InlineCode>{name}</InlineCode>
+      </H3>
+      <UL>
+        {children.map((enumValue: any) => (
+          <LI key={enumValue.name}>
+            <InlineCode>
+              {name}.{enumValue.name}
+            </InlineCode>
+            {comment ? ` - ${comment.shortText}` : null}
+          </LI>
+        ))}
+      </UL>
+    </>
+  );
+};
+
+const renderType = ({ name, comment, type }: any): JSX.Element => {
+  return (
+    <>
+      <H3>
+        <InlineCode>{name}</InlineCode>
+      </H3>
+      {comment ? (
+        <P>
+          <ReactMarkdown renderers={renderers}>{comment.shortText}</ReactMarkdown>
+        </P>
+      ) : null}
+      <table>
+        <thead>
+          <tr>
+            <th>Name</th>
+            <th>Type</th>
+            <th>Description</th>
+          </tr>
+        </thead>
+        <tbody>
+          {type.declaration.children.map((typeProperty: any) => (
+            <tr key={typeProperty.name}>
+              <td>
+                <B>{typeProperty.name}</B>
+                {typeProperty.flags?.isOptional ? (
+                  <>
+                    <br />
+                    <span css={STYLES_OPTIONAL}>(optional)</span>
+                  </>
+                ) : null}
+              </td>
+              <td>
+                <InlineCode>{resolveTypeName(typeProperty.type)}</InlineCode>
+              </td>
+              <td>{typeProperty.comment ? typeProperty.comment.shortText : '-'}</td>
+            </tr>
+          ))}
+        </tbody>
+      </table>
+    </>
+  );
+};
+
+const processData = (packageName: string, version: string = 'unversioned'): JSX.Element => {
+  try {
+    const data = require(`~/public/static/data/${version}/${packageName}.json`);
+
+    const methods = data.children.filter(g => g.kind === 64);
+    const types = data.children.filter(g => g.kind === 4194304);
+    const enums = data.children.filter(g => g.kind === 4);
```

Comment:
We definitely need to convert this numeric values into constants. Maybe these are available via some ts-doc-related package? 🤔 If no, we need to define them ourselves.
Best approach would be to have a method:
```
function getData(data: JSONData, dataType: DataTypeEnum): SpecificDataTypeSet
```

Reply:
There are types build in `typedoc` (`ReflectionKind`), but to utilize them we will need to import whole TypeDoc package as `dependency` to the `docs`which is a bit of overkill:
* https://github.com/TypeStrong/typedoc/blob/master/src/lib/models/reflections/abstract.ts#L37-L64

Other TypeDoc users seems also to skip this and creating there own types in-place:
* https://github.com/jquense/docpocalypse/blob/master/packages/theme/src/components/typedoc-types.ts#L4-L38

Reply:
Also the included in TypeDoc `JSONOutput` schema is very fragile and unpleasant to use, so for now (PoC) I have opted out dealing with this problem:
* https://github.com/TypeStrong/typedoc/issues/789
* https://github.com/TypeStrong/typedoc/issues/605